### PR TITLE
Initial work on Docker Repos support

### PR DESCRIPTION
**Pull request by jits**
Note: this is not yet functional:
- It lays out the barebones UI and hub backend for supporting Docker Repo management for projects.
- In upcoming work we'll hook this up to the backend integration service (via a message bus) and add access management.

### CODE REVIEW

File: platform-hub-api/app/models/ability.rb [hunk 1]
```diff
@@ -53,6 +53,16 @@ def initialize(user)
     end
 
 
+    # Docker repos in projects
+
+    can :read_docker_repos_in_project, Project do |project|
+      can_participate_in_project project, user
+    end
+    can :administer_docker_repos_in_project, Project do |project|
+      can_participate_in_project project, user
```
Comment (marcinc) [inline platform-hub-api/app/models/ability.rb]: This should be `can_administer_project` ?

File: platform-hub-api/app/models/docker_repo.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  before_validation :add_project_shortname_to_name
+
+  belongs_to :service, -> { readonly }
+
+  scope :by_services, ->(ids) { where(service_id: ids) }
+
+  enum status: {
+    pending: 'pending',
+    ready: 'ready',
+    deleting: 'deleting',
```
Comment (marcinc) [inline platform-hub-api/app/models/docker_repo.rb]: Just a thought. What if repo couldn't get created upstream for whatever reason? Should we add `failed` status (or similar) or we just keep it permanently in `pending` in such case?

File: platform-hub-api/app/models/docker_repo.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  def readonly?
+    if persisted?
+      read_only_attrs = self.class.readonly_attributes.to_a
+      if read_only_attrs.any? {|f| send(:"#{f}_changed?")}
+        raise ActiveRecord::ReadOnlyRecord, "#{read_only_attrs.join(', ')} can't be modified"
+      end
+    end
+  end
+
+  def add_project_shortname_to_name
```
Comment (marcinc) [inline platform-hub-api/app/models/docker_repo.rb]: maybe name it in a more generic way e.g.`build_docker_repo_name` ?

File: platform-hub-api/app/models/ability.rb (hunk 1 above)
Comment (jits) [reply platform-hub-api/app/models/ability.rb]: I think the [consensus in #hub_ecr](https://hod-dsp.slack.com/archives/CDMFKN5TJ/p1541601224025900) was that any one in a project can create or delete repos.

File: platform-hub-api/app/models/docker_repo.rb (hunk 2 above)
Comment (marcinc) [reply platform-hub-api/app/models/docker_repo.rb]: Cool 👍

**Pull request merged**